agr_literature_service/lit_processing/pdf2tei/pdf2tei.py

```diff
@@ -0,0 +1,71 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+def main():
+    engine = create_engine(SQLALCHEMY_DATABASE_URL, connect_args={"options": "-c timezone=utc"})
+    new_session = sessionmaker(bind=engine, autoflush=True)
+    db = new_session()
+    mod_abbreviations = [mod.abbreviation for mod in db.query(ModModel.abbreviation).all()]
+    for mod_abbreviation in mod_abbreviations:
+        for ref_id in get_ref_ids_with_workflow_status(db=db, workflow_atp_id="ATP:0000162", mod_abbreviation=mod_abbreviation):
+            reference_obj = db.query(ReferenceModel).filter(ReferenceModel.reference_id == ref_id).one()
+            ref_file_id_to_convert = get_main_pdf_referencefile_id(db=db, curie_or_reference_id=ref_id,
+                                                                   mod_abbreviation=mod_abbreviation)
+            if ref_file_id_to_convert:
+                ref_file_obj: ReferencefileModel = db.query(ReferencefileModel).filter(
+                    ReferencefileModel.referencefile_id == ref_file_id_to_convert).one()
+                file_content = download_file(db=db, referencefile_id=ref_file_id_to_convert,
+                                             mod_access=OktaAccess.ALL_ACCESS, use_in_api=False)
+                # Define the GROBID API endpoint
+                grobid_api_url = os.environ.get("PDF2TEI_API_URL",
+                                                "https://grobid.alliancegenome.org/api/processFulltextDocument")
+
+                # Send the file content to the GROBID API
+                response = requests.post(grobid_api_url, files={'input': ("file", file_content)})
+
+                # Check the response
+                if response.status_code == 200:
+                    logger.info(f"referencefile with ID {str(ref_file_id_to_convert)} successfully processed by GROBID.")
+                    metadata = {
+                        "reference_curie": reference_obj.curie,
+                        "display_name": ref_file_obj.display_name,
+                        "file_class": "tei",
+                        "file_publication_status": "final",
+                        "file_extension": "tei",
+                        "pdf_type": None,
+                        "is_annotation": None,
+                        "mod_abbreviation": mod_abbreviation
+                    }
+                    file_upload(db=db, metadata=metadata, file=UploadFile(file=BytesIO(response.content),
+                                                                          filename=ref_file_obj.display_name),
+                                upload_if_already_converted=True)
+                    transition_to_workflow_status(db=db, curie_or_reference_id=str(reference_obj.reference_id),
+                                                  mod_abbreviation=mod_abbreviation,
+                                                  new_workflow_tag_atp_id="ATP:0000163", transition_type="automated")
```

Comment:
Not sure about hardcoding the success and fails. I think it might be better to store these in the conditions (see file upload as an example). 
So calling job_change_atp_code(db: Session, reference_workflow_tag_id: int, condition: str = ""):
 with condition as 'on_success' or 'on_failed' as appropriate.